core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -173,6 +174,24 @@ private[scheduler] class BlacklistTracker (
         listenerBus.post(SparkListenerExecutorBlacklisted(now, exec, newTotal))
         executorIdToFailureList.remove(exec)
         updateNextExpiryTime()
+        // Add executor to blacklist before attempting to kill it. This allows a scheduler backend
+        // to immediately fail to allocate resources on this executor, since killing could be
+        // asynchronous.
```

Comment:
This comment is only needed above the nodeBlacklist case.  Each executor ID is always unique, so we dont' need to worry about the cluster manager registering the same executor again.

Also, I think "allows ... to fail" is a little confusing.  How about "This is so the scheduler backend can reject any additional resources the cluster manager gives on the blacklisted node."